### PR TITLE
Add maintenance job

### PR DESCRIPTION
**Pull request by zugao**
This PR adds Redis maintenance support.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: class/defaults.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ parameters:
       appcat:
         registry: ghcr.io
         repository: vshn/appcat
-        tag: v4.8.0
+        tag: add-maintenance-job
```
Comment (zugao) [inline class/defaults.yml:31]: This will change to v4.9.0

File: component/provider.jsonnet [hunk 2]
```diff
@@ -178,6 +178,11 @@ local controllerConfigRef(config) =
           resources: [ 'schedules' ],
           verbs: [ 'get', 'list', 'watch', 'update', 'patch', 'create', 'delete' ],
         },
+        {
```
Comment (Kidswiss) [inline component/provider.jsonnet:181]: I'm a bit confused why you add this rule to the provider-kubernetes? We directly apply the `Release` object without going through provider-kubernetes. So provider-kubernetes should never touch this, or am I missing something?

**Review by Kidswiss — CHANGES_REQUESTED** [review]

File: component/provider.jsonnet (hunk 2 above)
Comment (zugao) [reply component/provider.jsonnet:181]: yeah it seems you are right I will remove this.

**Review by Kidswiss — APPROVED** [review]
Please don't forget to adjust the tag :)

**Pull request merged**